### PR TITLE
Share Steam library directories when starting singleplayer game

### PR DESCRIPTION
**Pull request by kakurasan**
See https://github.com/truckersmp-cli/truckersmp-cli/pull/159#issuecomment-846528645

Depends on #181

### CODE REVIEW

**Comment by Lucki** [comment]
FYI: Today SCS dropped a beta for native multiplayer coming in 1.41: https://forum.scssoft.com/viewtopic.php?f=214&t=300266
We drove some jobs together and while it still has some limitations it's already working great even across platforms.

File: truckersmp_cli/utils.py [hunk 1]
```diff
@@ -353,6 +353,42 @@ def get_short_size(size_bytes):
...
+    # example:
+    #     "LibraryFolders"
+    #     {
+    #         "TimeNextStatsReport"  "xxxx"
+    #         "ContentStatsID"       "xxxx"
+    #         "1"                    "/path/to/steam/library1"
+    #         "2"                    "/path/to/steam/library2"
+    #     }  -> [steamdir, "/path/to/steam/library1", "/path/to/steam/library2"]
+    try:
+        with open(os.path.join(steamdir, File.steamlibvdf_inner)) as f_vdf:
```
Comment (Lucki) [inline truckersmp_cli/utils.py:375]: Any particular reason you aren't using python-vdf here?

File: truckersmp_cli/utils.py [hunk 2]
```diff
@@ -353,6 +353,42 @@ def get_short_size(size_bytes):
...
 
 
+def get_steam_library_dirs(steamdir):
+    """
+    Get Steam library directories.
+
+    steamdir: Steam installation path
+    """
+    # the first Steam library directory is the Steam installation
+    steam_libraries = [steamdir, ]
```
Comment (Lucki) [inline truckersmp_cli/utils.py:363]: Isn't this ?
```suggestion
    # the first Steam library directory is the Steam installation
    steam_libraries = [os.path.join(steamdir, "steamapps"), ]
```

Beware older installations may still have `os.path.join(steamdir, "SteamApps")`

File: truckersmp_cli/main.py [hunk 3]
```diff
@@ -222,7 +222,9 @@ def start_with_proton():
         logging.info("Using Steam Runtime container")
         # share directories with Steam Runtime container
         shared_paths = [Args.gamedir, Args.protondir, Args.prefixdir]
-        if not Args.singleplayer:
+        if Args.singleplayer:
+            shared_paths += get_steam_library_dirs(steamdir)
```
Comment (Lucki) [inline truckersmp_cli/main.py:226]:
```suggestion
        if Args.singleplayer:
            # workshop mods may be loaded from other steam libraries
            # despite they are also present in our gamedir library
            shared_paths += get_steam_library_dirs(steamdir)
```

File: truckersmp_cli/variables.py [hunk 4]
```diff
@@ -51,6 +51,7 @@ class File:
     """Files."""
 
     loginvdf_inner = "config/loginusers.vdf"
+    steamlibvdf_inner = "steamapps/libraryfolders.vdf"
```
Comment (Lucki) [inline truckersmp_cli/variables.py:54]: Beware that older installations may still have `SteamApps`.

Valve changed that for new installations to all lowercase some time ago but I'm not sure if they transitioned existing installations.

File: truckersmp_cli/utils.py (hunk 1 above)
Comment (kakurasan) [reply truckersmp_cli/utils.py]: Yes.

* We don't need to *parse VDF*
    * It's enough to read each line and check the 2 quoted string
    * `get_steam_library_dirs()` is already simple
* I would not like to add a "required" dependency
    * `get_steam_library_dirs()` would be a part of "basic" feature

File: truckersmp_cli/main.py (hunk 3 above)
Comment (kakurasan) [reply truckersmp_cli/main.py:228]: Feel free to commit the suggestion when the others are ready.

**Review by Lucki — APPROVED** [review]

File: truckersmp_cli/utils.py [hunk 5]
```diff
@@ -354,6 +354,48 @@ def get_short_size(size_bytes):
...
+    # additional directories are stored in libraryfolders.vdf
+    #
+    # example:
+    #     "LibraryFolders"
+    #     {
+    #         "TimeNextStatsReport"  "xxxx"
+    #         "ContentStatsID"       "xxxx"
+    #         "1"                    "/path/to/steam/library1"
+    #         "2"                    "/path/to/steam/library2"
+    #     }  -> [steamdir, "/path/to/steam/library1", "/path/to/steam/library2"]
```
Comment (Lucki) [inline truckersmp_cli/utils.py:376]: https://github.com/tkashkin/GameHub/issues/558#issuecomment-873150605

**Pull request merged**